sdk/storage/azure-storage-queues/inc/azure/storage/queues/queue_service_client.hpp

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include <azure/core/credentials/credentials.hpp>
+#include <azure/storage/common/storage_credential.hpp>
+
+#include "azure/storage/queues/queue_client.hpp"
+#include "azure/storage/queues/queue_options.hpp"
+#include "azure/storage/queues/queue_responses.hpp"
+
+namespace Azure { namespace Storage { namespace Queues {
+
+  class QueueServiceClient final {
+  public:
+    static QueueServiceClient CreateFromConnectionString(
+        const std::string& connectionString,
+        const QueueClientOptions& options = QueueClientOptions());
+
+    explicit QueueServiceClient(
+        const std::string& serviceUrl,
+        std::shared_ptr<StorageSharedKeyCredential> credential,
+        const QueueClientOptions& options = QueueClientOptions());
+
+    explicit QueueServiceClient(
+        const std::string& serviceUrl,
+        std::shared_ptr<Core::Credentials::TokenCredential> credential,
+        const QueueClientOptions& options = QueueClientOptions());
+
+    explicit QueueServiceClient(
+        const std::string& serviceUrl,
+        const QueueClientOptions& options = QueueClientOptions());
```

Comment:
still curious about this.  No-credential client?

Reply:
This is correct. No-credential client.